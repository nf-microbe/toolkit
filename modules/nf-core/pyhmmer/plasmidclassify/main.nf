process PYHMMER_PLASMIDCLASSIFY {
    tag "${meta.id}"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/pyhmmer_pandas:9f16bf144bcd5b1a':
        'community.wave.seqera.io/library/pyhmmer_pandas:842847931f77e2ab' }"

    input:
    tuple val(meta), path(faa)
    path plasmid_hmms

    output:
    tuple val(meta), path("${prefix}.plasmid_hmms.tsv")     , emit: hits
    tuple val(meta), path("${prefix}.plasmid_markers.tsv")  , emit: markers
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    pyhmmer_plasmidclassify.py \\
        --input ${faa} \\
        --plasmid_hallmarks ${plasmid_hmms} \\
        --prefix ${prefix} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        pandas: \$(python -c "import pandas; print(pandas.__version__)")
        pyhmmer: \$(python -c "import pyhmmer; print(pyhmmer.__version__)")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.plasmid_hmms.tsv
    touch ${prefix}.plasmid_markers.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        pandas: \$(python -c "import pandas; print(pandas.__version__)")
        pyhmmer: \$(python -c "import pyhmmer; print(pyhmmer.__version__)")
    END_VERSIONS
    """
}
