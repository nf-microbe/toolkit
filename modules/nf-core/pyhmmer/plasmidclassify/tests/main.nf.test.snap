{
    "bacteroides_fragilis - faa - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.plasmid_hmms.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.plasmid_markers.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,ef5266117af8b0677330ad16f541b552"
                ],
                "hits": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.plasmid_hmms.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "markers": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.plasmid_markers.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ef5266117af8b0677330ad16f541b552"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-12T15:15:08.04115751"
    },
    "bacteroides_fragilis - faa": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.plasmid_hmms.tsv:md5,9fac5cb84608517ea940595f275d65e8"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.plasmid_markers.tsv:md5,cd9749ea509af4ba03535150b2cd2d75"
                    ]
                ],
                "2": [
                    "versions.yml:md5,ef5266117af8b0677330ad16f541b552"
                ],
                "hits": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.plasmid_hmms.tsv:md5,9fac5cb84608517ea940595f275d65e8"
                    ]
                ],
                "markers": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.plasmid_markers.tsv:md5,cd9749ea509af4ba03535150b2cd2d75"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ef5266117af8b0677330ad16f541b552"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-12T15:14:43.667404897"
    }
}